RecoLocalTracker/SiPixelRecHits/interface/SiPixelTemplate2D.h

```diff
@@ -134,35 +174,62 @@ class SiPixelTemplate2D {
    float qscale() {return qscale_;}    //!< charge scaling factor
    float s50() {return s50_;}          //!< 1/2 of the pixel threshold signal in adc units
    float sxymax() {return sxymax_;}    //!< max pixel signal for pixel error calculation
-   float xytemp(int j, int i) {        //!< get the 2-d template for pixel j (x), i (y) in BXM2 x BYM2 array (x,y)=(0,0) in lower LH corner of pixel[1][1]
+   float scalex(int i) {
 #ifndef SI_PIXEL_TEMPLATE_STANDALONE
```

Comment:
For the ease of reading (and maintenance, see here), couldn't this check of data corruptness be separated in a function(i,"function_name",imax) and called in a single line within all these methods?

Reply:
I'm not sure I'm following.  Which line is this comment about?

Also, this is Morris's code: I'll need to ask him.

Reply:
e.g. my suggestion is that you could define
```
bool checkIllegalIndex(const std::string whichMethod, int indMax,i int i) 
{
#ifndef SI_PIXEL_TEMPLATE_STANDALONE
      if(i < 0 || i > indMax) {throw cms::Exception("DataCorrupt") << "SiPixelTemplate2D::"<<whichMethod<<" called with illegal index = " << i << std::endl;}
#else
      assert(i>=0 && i<indMax+1);
#endif
return true,
}
```
and then inside every method you can simply call (e.g.)
```
float scalex(int i) {
   if (checkIllegalIndex("scalex",3,i) return scalex_[i];   //!< x-error scale factor in 4 charge bins
}
```
which is more easily readable (and possible future maintenance is limited to one place).
Moreover, would the argument ```i``` be a ```unsigned int``` (as it would be normal for indeces) everything could even more compact, because no check for it being positive would be needed